pkg/ovirt/install.sh

```diff
@@ -0,0 +1,103 @@
+#
+# Main task: update configuration files for Engine URL
+# Reason: Engine URL can't be determined from the VDSM host automatically, so it must be provided by user.
+#
+# What it does:
+#      update content-security-policy in the 'ovirt' Cockpit plugin
+#      update cockpit-ovirt runtime configuration (to assemble oVirt REST API URL)
+#
+# How to run:
+#      either manually after rpm installations as root:
+#         # cd [COCKPIT_OVIRT_INSTALL_DIR] && ./install.sh https://[ENGINE_HOST]/ovirt-engine/
+#         # "COCKPIT_OVIRT_INSTALL_DIR" usually refers to /usr/share/cockpit/ovirt
+#
+#      or
+#         login into cockpit as the 'root' user
+#         access the 'ovirt' plugin
+#         installation dialog shows up to handle the install.sh script execution from UI
+#
+
+ENGINE_URL=$1
+COCKPIT_DIR=${2:-/usr/share/cockpit}  # TODO: get it dynamically via 'rpm -q cockpit-shell --fileprovide' or 'dpkg -L'
+ETC_COCKPIT_DIR='/etc/cockpit'
+VIRSH_CONNECTION_URI=$3 # optional
+VIRSH_CONNECTION_NAME='remote' # used if VIRSH_CONNECTION_URI is set
+
+EXIT_PARAMS=1 # wrong command parameters
+EXIT_NO_ACCESS_MACHINES_OVIRT_CONFIG=3 # can't write to /etc/cockpit/machines-ovirt.config, try as root
+EXIT_NO_ACCESS_OVIRT_OVERRIDE=4 # can't write to cockpit/ovirt/override.json
+
+function usage() {
+  echo Usage: $0 '[ENGINE_URL] [[COCKPIT_INSTLLATION_DIR]] [[VIRSH_CONNECTION_URI]]'
+  echo Example: $0 https://engine.mydomain.com/ovirt-engine/
+  echo Example: $0 https://engine.mydomain.com/ovirt-engine/ /usr/share/cockpit
+  echo Example: $0 https://engine.mydomain.com/ovirt-engine/ /usr/share/cockpit qemu:///system
+}
+
+function checkParams() {
+  if [ x${ENGINE_URL} = x ] ; then
+    usage
+    exit ${EXIT_PARAMS}
+  else
+    echo Registering for ENGINE_URL: ${ENGINE_URL}
+  fi
+}
+
+function generateProviderConfig() {
+  CONFIG_FILE=${ETC_COCKPIT_DIR}/machines-ovirt.config
+
+  # TODO: decrease polling interval if AuditLog-based updates proof to be working
+  echo "{ \
+      \"debug\": false, \
+      \"ovirt_polling_interval\": 60000, \
+      \"cockpitPort\": 9090, \
+      \"OVIRT_BASE_URL\": \"$ENGINE_URL\"," \
+        > ${CONFIG_FILE} || exit ${EXIT_NO_ACCESS_MACHINES_OVIRT_CONFIG}
+
+  if [ x${VIRSH_CONNECTION_URI} = x ] ; then
+    echo " \
+        \"Virsh\": { \
+            \"connections\": { \
+                \"system\": { \
+                    \"params\": [\"-c\", \"qemu:///system\"] \
+                }, \
+                \"session\": { \
+                    \"params\": [\"-c\", \"qemu:///session\"] \
+                } \
+            } \
+        }" >> ${CONFIG_FILE}
+  else
+    echo " \
+        \"Virsh\": { \
+            \"connections\": { \
+                \"${VIRSH_CONNECTION_NAME}\": { \
+                    \"params\": [\"-c\", \"${VIRSH_CONNECTION_URI}\"] \
+                } \
+            } \
+        }" >> ${CONFIG_FILE}
+  fi
```

Comment:
What local command line tooling is used to interact with oVirt. What commands are used to set it up? Can we just share configuration with the local tooling?

Reply:
To run pkg/ovirt, tooling used by pkg/machines is sufficient (recently means `virsh`).

This configuration is for hacking only, like within integration tests where VDSM/Libvirt is running on a different machine than the one with cockpit. So for the test purposes, `virsh` connects to the ovirt-host via `qemu+tcp:///...` URI while leveraging the fact that an oVirt host has Libvirt port open by default due to VM migrations.

It's not a recent use case, but I can imagine weird deployments where the user might want manage VMs running on a different host than the one with cockpit. Its would be feasible by providing this configuration.

Within https://github.com/cockpit-project/cockpit/issues/7484, such a configuration will be available even for pkg/machines.